python/lsst/daf/butler/datastores/fileLikeDatastore.py

```diff
@@ -334,26 +421,48 @@ def _get_dataset_location_info(self, ref):
 
         return location, storedFileInfo
 
-    def _can_remove_dataset_artifact(self, ref):
+    def _get_dataset_locations_info(self, ref):
+        r"""Find all the `Location`\ s  of the requested dataset in the
+        `Datastore` and the associated stored file information.
+
+        Parameters
+        ----------
+        ref : `DatasetRef`
+            Reference to the required `Dataset`.
+
+        Returns
+        -------
+        results : `list` [`tuple` [`Location`, `StoredFileInfo` ]]
+            Location of the dataset within the datastore and
+            stored information about each file and its formatter.
+        """
+        # Get the file information (this will fail if no file)
+        records = self.getStoredItemsInfo(ref)
+
+        # Use the path to determine the location
+        return [(self.locationFactory.fromPath(r.path), r) for r in records]
+
+    def _can_remove_dataset_artifact(self, ref, location):
```

Comment:
I assume this function only exists for cases like DECam raws, where there are multiple top-level datasets associated with the same artifact, not anything to do with composites?

Reply:
It was there previously  (I think somewhere else) and is for DECam-like things but also to do with composites when they are stored in the registry. If you wanted to delete a component I had to make sure you didn't delete the file if there were multiple components associated with one file.